sklearn/tests/random_seed.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+from os import environ
+from random import Random
+from datetime import datetime
+
+# Definition of the random_seed fixture. See the docstring of the fixture for
+# more information.
+
+RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+random_seed_var = environ.get("SKLEARN_TESTS_GLOBAL_RANDOM_SEED")
+if random_seed_var is None:
+    # This is the way.
+    random_seeds = [42]
+elif random_seed_var == "any":
+    # Pick-up one seed at random in the range of admissible random seeds.
+    #
+    # Note, to make sure that all pytest-xdist workers see the same seed, we
+    # seed the meta-random number generator with a value derived either from a
+    # CI build number variables, or from the year and the day for local builds
+    try:
+        meta_seed = int(environ["BUILD_NUMBER"])
```

Comment:
This is very specific to Azure Pipelines. I'm okay with merging this PR as is.

As a follow up PR, I think there is a way to generate the seeds on the `main` process and pass it to the workers for xdist.

Reply:
I opened https://github.com/ogrisel/scikit-learn/pull/12 to demonstrate generating the seed and passing it to the workers.

Reply:
Thanks!